### PR TITLE
ARM Templates to Copy data from Azure Blob Storage to Azure Database for PostgreSQL

### PR DESCRIPTION
**Pull request by ramnov**
ARM Templates to Copy data from Azure Blob Storage to Azure Database for PostgreSQL

### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-data-factory-v2-blob-to-postgresql-copy/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,205 @@
+{
+  "contentVersion": "1.0.0.0",
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "parameters": {
+    "dataFactoryName": {
+      "type": "string",
```
Comment (bmoore-msft) [inline 101-data-factory-v2-blob-to-postgresql-copy/azuredeploy.json]: add a globally unique defaultValue using uniqueString

File: 101-data-factory-v2-blob-to-postgresql-copy/prereqs/prereq.azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
+{
```
Comment (bmoore-msft) [inline 101-data-factory-v2-blob-to-postgresql-copy/prereqs/prereq.azuredeploy.json]: doesn't look like you're actually using these - is that intentional?

File: 101-data-factory-v2-blob-to-postgresql-copy/azuredeploy.json (hunk 1 above)
Comment (ramnov) [reply 101-data-factory-v2-blob-to-postgresql-copy/azuredeploy.json]: sure

File: 101-data-factory-v2-blob-to-postgresql-copy/prereqs/prereq.azuredeploy.json (hunk 2 above)
Comment (ramnov) [reply 101-data-factory-v2-blob-to-postgresql-copy/prereqs/prereq.azuredeploy.json]: Ideally, we are supposed to use outputs from prereq 'AzureStorage_connectionString' and 'AzurePostgreSqlDatabase_connectionString'. But the PR check always fails with message - output should not contain secrets. So removed the outputs. 

The following ARM template does have secrets in output : 
https://github.com/Azure/azure-quickstart-templates/blob/master/101-data-factory-v2-blob-to-sql-copy/prereqs/prereq.azuredeploy.json

Comment (ramnov) [reply 101-data-factory-v2-blob-to-postgresql-copy/prereqs/prereq.azuredeploy.json]: @bmoore-msft : I have updated the ARM Templates using your suggestion to pass storage account resource id.

File: 101-data-factory-v2-blob-to-postgresql-copy/azuredeploy.json [hunk 3]
```diff
@@ -53,6 +73,8 @@
     }
   },
   "variables": {
+    "storageAccountName": "[last(split(parameters('storageAccountResourceId'), '/'))]",
+    "azurePostgreSqlDatabaseConnectionString": "[concat('Server=', parameters('postgreSqlServerName'),'.postgres.database.azure.com;Port=5432;Database=' ,parameters('postgreSqlDatabase'), ';UID=', parameters('postgreSqlUserName'), '@', parameters('postgreSqlServerName'), ';Password=', parameters('postgreSqlPassword'), ';SSL Mode=Require')]",
```
Comment (bmoore-msft) [inline 101-data-factory-v2-blob-to-postgresql-copy/azuredeploy.json]: use reference to get the FQDN - you'll need to move this from variables to where it's used below (reference cannot be used in a variable)

Comment (ramnov) [reply 101-data-factory-v2-blob-to-postgresql-copy/azuredeploy.json]: sure. Used reference

Comment (bmoore-msft) [reply 101-data-factory-v2-blob-to-postgresql-copy/azuredeploy.json]: This still needs to be removed (and the one instance below updated...

Comment (ramnov) [reply 101-data-factory-v2-blob-to-postgresql-copy/azuredeploy.json]: I have indeed removed it.

**Pull request merged**